modules/repository-s3/src/main/java/org/elasticsearch/repositories/s3/S3Service.java

```diff
@@ -124,24 +127,44 @@ class S3Service extends AbstractLifecycleComponent {
     final TimeValue compareAndExchangeTimeToLive;
     final TimeValue compareAndExchangeAntiContentionDelay;
     final boolean isStateless;
+    @Nullable // if multi-project is disabled
+    private final S3PerProjectClientManager s3PerProjectClientManager;
 
     S3Service(
         Environment environment,
-        Settings nodeSettings,
+        ClusterService clusterService,
+        ProjectResolver projectResolver,
         ResourceWatcherService resourceWatcherService,
         Supplier<Region> defaultRegionSupplier
     ) {
+        final Settings settings = clusterService.getSettings();
```

Comment:
nit: I liked the name `nodeSettings` in this context, it can otherwise be confusing whether we're talking about node-level settings or repository settings from the `RepositoryMetadata`.

Reply:
9bc5d0939df608d9c3b372024535ca3546b59b33